tests/publiccloud/img_proof.pm

```diff
@@ -27,33 +27,46 @@ our $gce_byos      = 'test_sles_wait_on_registration,test_sles,test_sles_gce';
 our $gce_on_demand = 'test_sles_wait_on_registration,test_sles,test_update,test_sles_smt_reg,test_sles_guestregister,test_sles_on_demand,test_sles_gce';
 
 our $img_proof_tests = {
-    'Azure-BYOS'       => $azure_byos,
-    'Azure-Basic'      => $azure_on_demand,
-    'Azure-Standard'   => $azure_on_demand,
-    'Azure-CHOST-BYOS' => $azure_byos,
-    'Azure-HPC'        => $azure_on_demand,
-    'Azure-HPC-BYOS'   => $azure_byos,
-
-    'EC2-CHOST-BYOS'   => $ec2_byos_chost,
-    'EC2-HVM'          => $ec2_on_demand,
-    'EC2-HVM-ARM'      => $ec2_on_demand,
-    'EC2-HVM-BYOS'     => $ec2_byos,
-    'EC2-HVM-HPC-BYOS' => $ec2_byos,
-
-    GCE              => $gce_on_demand,
-    'GCE-BYOS'       => $gce_byos,
-    'GCE-CHOST-BYOS' => $gce_byos,
+    'Azure-BYOS'         => $azure_byos,
+    'AZURE-BYOS-Updates' => $azure_byos,
+    'Azure-Basic'        => $azure_on_demand,
+    'Azure-Standard'     => $azure_on_demand,
+    'Azure-CHOST-BYOS'   => $azure_byos,
+    'Azure-HPC'          => $azure_on_demand,
+    'Azure-HPC-BYOS'     => $azure_byos,
+
+    'EC2-CHOST-BYOS'       => $ec2_byos_chost,
+    'EC2-HVM'              => $ec2_on_demand,
+    'EC2-HVM-ARM'          => $ec2_on_demand,
+    'EC2-HVM-BYOS'         => $ec2_byos,
+    'EC2-HVM-BYOS-Updates' => $ec2_byos,
+    'EC2-HVM-HPC-BYOS'     => $ec2_byos,
+
+    GCE                => $gce_on_demand,
+    'GCE-BYOS'         => $gce_byos,
+    'GCE-BYOS-Updates' => $gce_byos,
+    'GCE-CHOST-BYOS'   => $gce_byos,
 };
 
 sub run {
-    my ($self) = @_;
+    my ($self, $args) = @_;
 
     $self->select_serial_terminal;
 
-    my $flavor   = get_required_var('FLAVOR');
-    my $provider = $self->provider_factory();
-    my $instance = $provider->create_instance();
-    my $tests    = get_required_var('PUBLIC_CLOUD_IMG_PROOF_TESTS');
+    my $flavor = get_required_var('FLAVOR');
+    my $tests  = get_required_var('PUBLIC_CLOUD_IMG_PROOF_TESTS');
+    my $provider;
+    my $instance;
+
+    # QAM passes the instance as argument
+    if (get_var('PUBLIC_CLOUD_QAM')) {
+        $instance         = $args->{my_instance};
+        $provider         = $args->{my_provider};
+        $self->{provider} = $args->{my_provider};    # required for cleanup
```

Comment:
```suggestion
        $self->{provider} = $args->{my_provider};    # required for cleanup
        select_host_console();
```

Reply:
We are seeing thing failure: http://phoenix-openqa.qam.suse.de/tests/3143#step/img_proof/52 
It's because we run `$self->provider_factory();` in ssh_interactive_init on a different console.
This provider_factory for azure does a `azure login` (see `lib::publiccloud::azure::az_login()`), but this only the environment  to that console.

we might move the `$self->select_serial_terminal;` also down in that `else` block.

Reply:
Done.